Included Rust sources don't get highlighted

When importing a Rust source file, it gets incorrectly detected as text and therefore syntax highlighting doesn't work. I've dug around and it looks like the `.rs` extension is found earlier in the [language map](https://github.com/blakeembrey/language-map/blob/master/languages.json), under the RenderScript entry with the corresponding `aceMode` set as `text`.
See an example in my fork here: https://github.com/zsiciarz/gitbook-plugin-include-codeblock/commit/41115eb30b5e0cf979f1409e51070bef80f8eb26, the test output is:

``````
± % npm test                                                                                                                            !10116

> gitbook-plugin-include-codeblock@1.6.0 test /home/zbyszek/Development/gitbook-plugin-include-codeblock
> mocha



  parse
    ✓ should return object for replace
    ✓ should translate elixir extensions
    1) should translate Rust extensions
    sliced text
      ✓ should return sliced object for replace
      ✓ should return sliced `start`- text
      ✓ should return sliced -`end` text
    containIncludeLabel
      ✓ support import
      ✓ support include
      ✓ support command split by space
      ✓ support command split by ,


  9 passing (184ms)
  1 failing

  1) parse should translate Rust extensions:

      AssertionError:   # test/parser-test.js:43

  assert.equal(result.replaced, expected)
               |      |         |        
               |      |         "> <a name=\"test.rs\" href=\"fixtures/test.rs\">test.rs</a>\n\n``` rust\nextern crate num;\n```"
               |      "> <a name=\"test.rs\" href=\"fixtures/test.rs\">test.rs</a>\n\n``` text\nextern crate num;\n```"
               Object{target:"[include](fixtures/test.rs)",replaced:"> <a name=\"test.rs\" href=\"fixtures/test.rs\">test.rs</a>\n\n``` text\nextern crate num;\n```"}

      + expected - actual

       > <a name="test.rs" href="fixtures/test.rs">test.rs</a>

      -``` text
      +``` rust
       extern crate num;
       ```

      at Function.decoratedAssert [as equal] (node_modules/empower/lib/decorate.js:42:30)
      at Context.<anonymous> (test/parser-test.js:43:16)



npm ERR! Test failed.  See above for more details
``````

 I'm not entirely sure how to fix this, as RenderScript is a perfectly valid option for .rs files. This is probably happening as well with for example .h files being detected as C headers, even if they contain C++ or ObjC code. Perhaps we could use some sort of language hint directly in the directive? Something like:

```
[include:1-4,rust](../src/day2.rs) 
```

--- a/src/parser.js
+++ b/src/parser.js
@@ -33,15 +33,24 @@ export function getSliceRange(label) {
 }
 
 export function lookupLanguageByExtension(ext) {
-  for (var key in language_map) {
-    if (language_map.hasOwnProperty(key) && language_map[key].extensions) {
-      if(language_map[key].extensions.indexOf(ext) !== -1) {
-        return language_map[key].aceMode;
-      }
-    }
-  }
-
-  return undefined;
+    let aceMode;
+    Object.keys(language_map).some(langKey => {
+        const extensions = language_map[langKey]["extensions"];
+        /* TODO: These lang has not extensions
+        Ant Build System
+        Isabelle ROOT
+        Maven POMAnt Build System
+         */
+        if (!extensions) {
+            return false;
+        }
+        return extensions.some(extension => {
+            if (ext === extension) {
+                aceMode = language_map[langKey]["aceMode"];
+            }
+        });
+    });
+    return aceMode;
 }
 
 export function getLang(filePath) {
@@ -62,12 +71,18 @@ ${slicedCode.trim()}
 }
 
 function sliceCode(code, start, end) {
-  if (start === '' && end === '') return code;
+    if (start === '' && end === '') {
+        return code;
+    }
 
-  var splitted = code.split('\n');
-  if (start === '') { start = 1; }
-  if (end === '') { end = splitted.length; }
-  return splitted.slice(start - 1, end).join('\n');
+    var splitted = code.split('\n');
+    if (start === '') {
+        start = 1;
+    }
+    if (end === '') {
+        end = splitted.length;
+    }
+    return splitted.slice(start - 1, end).join('\n');
 }
 
 export function parse(content, baseDir) {

--- a/test/fixtures/test.rs
+++ b/test/fixtures/test.rs
@@ -1,0 +1,1 @@
+extern crate num;

--- a/test/parser-test.js
+++ b/test/parser-test.js
@@ -29,6 +29,19 @@ describe("parse", function () {
             + '``` elixir\nIO.puts \"test\"\n```';
         assert.equal(result.replaced, expected);
     });
+    it("should translate Rust extensions", function () {
+        var exs_content = `
+        [include](fixtures/test.rs)
+        `;
+        var results = parse(exs_content, __dirname);
+        assert(results.length > 0);
+        var result = results[0];
+        assert.equal(result.target, "[include](fixtures/test.rs)");
+        var expected = '> <a name="test.rs" href="fixtures/test.rs">test.rs</a>\n'
+            + '\n'
+            + '``` rust\nextern crate num;\n```';
+        assert.equal(result.replaced, expected);
+    });
     context("sliced text", function () {
         it("should return sliced object for replace", function () {
             var multiLineContent = "[include:4-6, line.js](fixtures/line.js)";

--- a/test/parser-test.js
+++ b/test/parser-test.js
@@ -1,51 +1,53 @@
 // LICENSE : MIT
 "use strict";
 import assert from "power-assert"
-import {parse,containIncludeLabel} from "../src/parser"
+import {parse, containIncludeLabel} from "../src/parser"
 var content = `
 [include](fixtures/test.js)
 `;
 describe("parse", function () {
-    it("should return object for replace", function () {
-        var results = parse(content, __dirname);
-        assert(results.length > 0);
-        var result = results[0];
-        assert.equal(result.target, "[include](fixtures/test.js)");
-        var expected = '> <a name="test.js" href="fixtures/test.js">test.js</a>\n'
-            + '\n'
-            + '``` javascript\nconsole.log(\"test\");\n```';
-        assert.equal(result.replaced, expected);
-    });
-    it("should translate elixir extensions", function () {
-        var exs_content = `
+    context("translate lang", function () {
+        it("should return object for replace", function () {
+            var results = parse(content, __dirname);
+            assert(results.length > 0);
+            var result = results[0];
+            assert.equal(result.target, "[include](fixtures/test.js)");
+            var expected = '> <a name="test.js" href="fixtures/test.js">test.js</a>\n'
+                + '\n'
+                + '``` javascript\nconsole.log(\"test\");\n```';
+            assert.equal(result.replaced, expected);
+        });
+        it("should translate elixir extensions", function () {
+            var exs_content = `
         [include](fixtures/test.exs)
         `;
-        var results = parse(exs_content, __dirname);
-        assert(results.length > 0);
-        var result = results[0];
-        assert.equal(result.target, "[include](fixtures/test.exs)");
-        var expected = '> <a name="test.exs" href="fixtures/test.exs">test.exs</a>\n'
-            + '\n'
-            + '``` elixir\nIO.puts \"test\"\n```';
-        assert.equal(result.replaced, expected);
-    });
-    it("should translate Rust extensions", function () {
-        var exs_content = `
+            var results = parse(exs_content, __dirname);
+            assert(results.length > 0);
+            var result = results[0];
+            assert.equal(result.target, "[include](fixtures/test.exs)");
+            var expected = '> <a name="test.exs" href="fixtures/test.exs">test.exs</a>\n'
+                + '\n'
+                + '``` elixir\nIO.puts \"test\"\n```';
+            assert.equal(result.replaced, expected);
+        });
+        it("should translate Rust extensions", function () {
+            var exs_content = `
         [include](fixtures/test.rs)
         `;
-        var results = parse(exs_content, __dirname);
-        assert(results.length > 0);
-        var result = results[0];
-        assert.equal(result.target, "[include](fixtures/test.rs)");
-        var expected = '> <a name="test.rs" href="fixtures/test.rs">test.rs</a>\n'
-            + '\n'
-            + '``` rust\nextern crate num;\n```';
-        assert.equal(result.replaced, expected);
+            var results = parse(exs_content, __dirname);
+            assert(results.length > 0);
+            var result = results[0];
+            assert.equal(result.target, "[include](fixtures/test.rs)");
+            var expected = '> <a name="test.rs" href="fixtures/test.rs">test.rs</a>\n'
+                + '\n'
+                + '``` rust\nextern crate num;\n```';
+            assert.equal(result.replaced, expected);
+        });
     });
     context("sliced text", function () {
         it("should return sliced object for replace", function () {
             var multiLineContent = "[include:4-6, line.js](fixtures/line.js)";
-            var results = parse(multiLineContent , __dirname);
+            var results = parse(multiLineContent, __dirname);
             assert(results.length > 0);
             var result = results[0];
             assert.equal(result.target, multiLineContent);
@@ -60,7 +62,7 @@ describe("parse", function () {
         });
         it("should return sliced `start`- text", function () {
             var multiLineContent = "[include:9-, line.js](fixtures/line.js)";
-            var results = parse(multiLineContent , __dirname);
+            var results = parse(multiLineContent, __dirname);
             assert(results.length > 0);
             var result = results[0];
             assert.equal(result.target, multiLineContent);
@@ -74,7 +76,7 @@ describe("parse", function () {
         });
         it("should return sliced -`end` text", function () {
             var multiLineContent = "[include:-2, line.js](fixtures/line.js)";
-            var results = parse(multiLineContent , __dirname);
+            var results = parse(multiLineContent, __dirname);
             assert(results.length > 0);
             var result = results[0];
             assert.equal(result.target, multiLineContent);

